### PR TITLE
fix: resolve plugins from npm packages correctly

### PR DESCRIPTION
**Pull request by volodymyr-rutskyi**
## What/Why/How?

When the CLI is installed globally and tries to resolve a plugin from NPM package, it can't find it because by default the `require.resolve` tries to look for the package from the CLI installation location. This would cause errors like below even though the plugin package is installed in the project's `node_modules` directory:

```
Something went wrong when processing :

  - Failed to load plugin "@redocly/portal-plugin-async-api/plugin.js": Cannot find module '@redocly/portal-plugin-async-api/plugin.js'
```

Added the config directory as an additional path to look for NPM packages when resolving plugins.

## Reference

Reported in internal conversation

## Check yourself

- [ ] Code changed? - Tested with redoc/reference-docs/workflows (internal)
- [ ] All new/updated code is covered with tests
- [ ] New package installed? - Tested in different environments (browser/node)

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: packages/core/src/config/config-resolvers.ts [hunk 1]
```diff
@@ -139,7 +139,7 @@ export async function resolvePlugins(
         const absolutePluginPath = existsSync(maybeAbsolutePluginPath)
           ? maybeAbsolutePluginPath
           : // For plugins imported from packages specifically
-            require.resolve(plugin);
+            require.resolve(plugin, { paths: [configDir] });
```
Comment (adamaltman) [inline packages/core/src/config/config-resolvers.ts:142]: Is there one and only one config dir? What happens if there are multiple nested configs?

File: .changeset/strange-starfishes-press.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"@redocly/openapi-core": patch
+"@redocly/cli": patch
+---
+
+Fixed a bug where the plugins in external NPM packages could not be resolved if the CLI package was installed globally.
```
Comment (HCloward) [inline .changeset/strange-starfishes-press.md:6]:
```suggestion
Fixed an issue where the plugins in external NPM packages could not be resolved if the CLI package was installed globally.
```

File: packages/core/src/config/config-resolvers.ts (hunk 1 above)
Comment (volodymyr-rutskyi) [reply packages/core/src/config/config-resolvers.ts:142]: The `node_modules` directory will always be in the root of the project. If the `configDir` is a nested directory, the plugin will still get resolved.

File: __tests__/smoke-plugins/run-smoke.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+echo
+echo "Directory content:"
+ls -a
+echo
+
+# Install plugin package
+npm i
+
+# Install CLI globally
+npm i redocly-cli.tgz -g
```
Comment (tatomyr) [inline __tests__/smoke-plugins/run-smoke.sh:15]: Let's move it under package.json's dependencies.

File: .github/workflows/smoke.yaml [hunk 4]
```diff
@@ -247,3 +247,55 @@ jobs:
...
+    needs: prepare-smoke
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/cache@v3
+        with:
+          path: __tests__/smoke-plugins/
+          key: cache-${{ github.run_id }}-${{ github.run_attempt }}
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14
```
Comment (tatomyr) [inline .github/workflows/smoke.yaml:300]:
```suggestion
  run-smoke-plugins--npm--node-22:
    needs: prepare-smoke
    runs-on: ubuntu-latest
    steps:
      - uses: actions/cache@v3
        with:
          path: __tests__/smoke-plugins/
          key: cache-${{ github.run_id }}-${{ github.run_attempt }}
      - uses: actions/setup-node@v3
        with:
          node-version: 22
```

I think it makes more sense to test against the latest Node 😄.

File: __tests__/smoke/run-smoke.sh [hunk 5]
```diff
@@ -8,6 +8,9 @@ echo "Directory content:"
 ls -a
 echo
 
+# Install npm plugin
+npm i
```
[the remark this answers is not in the document]
Comment (volodymyr-rutskyi) [reply __tests__/smoke/run-smoke.sh]: Reverted this change

File: __tests__/smoke-plugins/run-smoke.sh (hunk 3 above)
Comment (volodymyr-rutskyi) [reply __tests__/smoke-plugins/run-smoke.sh:15]: To reproduce the original issue this PR fixes, the CLI needs to be installed globally.

File: .changeset/strange-starfishes-press.md (hunk 2 above)
Comment (volodymyr-rutskyi) [reply .changeset/strange-starfishes-press.md]: Applied the suggestion

File: .github/workflows/smoke.yaml (hunk 4 above)
Comment (volodymyr-rutskyi) [reply .github/workflows/smoke.yaml]: Sure. Added jobs for node 22 for all smoke tests

File: .github/workflows/smoke.yaml [hunk 6]
```diff
@@ -26,6 +26,49 @@ jobs:
           path: __tests__/smoke/
           key: cache-${{ github.run_id }}-${{ github.run_attempt }}
 
+  prepare-smoke-plugins:
```
Comment (tatomyr) [inline .github/workflows/smoke.yaml:29]: Can we split smokes and plugin-smokes into different workflows? I feel like they should be updated separately.

File: scripts/prepare-smoke.sh [hunk 7]
```diff
@@ -7,6 +7,7 @@ npm run webpack-bundle
 npm run pack:prepare
 
 cp ./redocly-cli.tgz ./openapi-core.tgz ./dist/bundle.js ./__tests__/smoke/
+cp ./redocly-cli.tgz ./openapi-core.tgz ./__tests__/smoke-plugins
```
Comment (tatomyr) [inline scripts/prepare-smoke.sh:10]: I'd move that to a separate script. It's being called twice anyway: https://github.com/Redocly/redocly-cli/pull/1749/files#diff-e5a87d0236e416c1fa636174d597bd7628a7434d6fe5bacbe4badd042b94804bR40

**Review by tatomyr — COMMENTED** [review]
Left an optional suggestion. Otherwise looks good.

File: .github/workflows/smoke.yaml (hunk 6 above)
Comment (volodymyr-rutskyi) [reply .github/workflows/smoke.yaml]: Done

File: scripts/prepare-smoke.sh (hunk 7 above)
Comment (volodymyr-rutskyi) [reply scripts/prepare-smoke.sh]: Created a separate script for plugins smoke preparation

**Review by tatomyr — APPROVED** [review]

**Review by HCloward — APPROVED** [review]

**Pull request merged**